pkg/api/pod/warnings.go

```diff
@@ -357,3 +364,142 @@ func warningsForWeightedPodAffinityTerms(terms []api.WeightedPodAffinityTerm, fi
 	}
 	return warnings
 }
+
+// warningsForOverlappingVirtualPaths validates that there are no overlapping paths in single ConfigMapVolume, SecretVolume, DownwardAPIVolume and ProjectedVolume.
+// A volume can try to load different keys to the same path which will result in overwriting of the value from the latest registered key
+// Another possible scenario is when one of the path contains the other key path. Example:
+// configMap:
+//
+//		name: myconfig
+//		items:
+//		  - key: key1
+//		    path: path
+//	      - key: key2
+//			path: path/path2
+//
+// In such cases we either get `is directory` or 'file exists' error message.
+func warningsForOverlappingVirtualPaths(volumes []api.Volume) []string {
+	warnings := make([]string, 0)
+
+	for _, v := range volumes {
+		if v.ConfigMap != nil && v.ConfigMap.Items != nil {
+			w := checkVolumeMappingForOverlap(extractPaths(v.ConfigMap.Items), fmt.Sprintf("volume %q (ConfigMap %q): overlapping path", v.Name, v.ConfigMap.Name))
+			if len(w) > 0 {
+				warnings = append(warnings, w...)
+			}
+		}
+
+		if v.Secret != nil && v.Secret.Items != nil {
+			w := checkVolumeMappingForOverlap(extractPaths(v.Secret.Items), fmt.Sprintf("volume %q (Secret %q): overlapping path", v.Name, v.Secret.SecretName))
+			if len(w) > 0 {
+				warnings = append(warnings, w...)
+			}
+		}
+
+		if v.DownwardAPI != nil && v.DownwardAPI.Items != nil {
+			w := checkVolumeMappingForOverlap(extractPathsDownwardAPI(v.DownwardAPI.Items), fmt.Sprintf("volume %q (DownwardAPI): overlapping path", v.Name))
+			if len(w) > 0 {
+				warnings = append(warnings, w...)
+			}
+		}
+
+		if v.Projected != nil {
+			var sourcePaths, allPaths []string
+			var errorMessage string
+
+			for _, source := range v.Projected.Sources {
+				switch {
```

Comment:
Maybe comment that exactly one of these is set per Source -- I had to go convince myself of that.

Reply:
In convincing myself, I found that we allow 0 sources to be specified - I feel like that should have been an error, but the best we can do now is a warning -- want to add it?

Reply:
Sure